platform/security/idp/security-idp-server/src/main/java/org/codice/ddf/security/idp/server/IdpEndpoint.java

```diff
@@ -232,6 +262,178 @@ public void init() {
         new ResponseUnmarshaller());
   }
 
+  @Override
+  public Map<String, Set<String>> getActiveSessions() {
+    try {
+      return Security.getInstance().runWithSubjectOrElevate(this::getActiveSessionsSecure);
+    } catch (SecurityServiceException | InvocationTargetException e) {
+      SecurityLogger.audit("Failed to run command; insufficient permissions");
+    }
+
+    return Collections.emptyMap();
+  }
+
+  @VisibleForTesting
+  Map<String, Set<String>> getActiveSessionsSecure() {
+    return cookieCache.getAllSamlSubjects(securityManager);
+  }
+
+  @Override
+  public void invalidateSession(String subjectName) {
+    try {
+      Security.getInstance()
+          .runWithSubjectOrElevate(
+              (Callable<Void>)
+                  () -> {
+                    invalidateSessionSecure(subjectName);
+                    return null;
+                  });
+    } catch (SecurityServiceException | InvocationTargetException e) {
+      SecurityLogger.audit("Failed to run command; insufficient permissions");
+    }
+  }
+
+  @VisibleForTesting
+  void invalidateSessionSecure(String subjectName) {
+    String cacheKey = cookieCache.getCacheKeyBySubjectName(subjectName, securityManager);
+    if (cacheKey == null) {
+      LOGGER.debug(
+          "No cache element found for subject name {}; skipping invalidation.", subjectName);
+      return;
+    }
+
+    LogoutState logoutState = new LogoutState(getActiveSps(cacheKey));
+
+    // TODO: 1/3/18 sufficient?
+    logoutState.setOriginalIssuer("");
+
+    logoutState.setNameId(subjectName);
+    logoutState.setOriginalRequestId(UUID.randomUUID().toString());
+
+    String relayState = UUID.randomUUID().toString();
```

Comment:
Likewise, there is no use for relay state over the SOAP binding for an IdP-initiated logout. There's no input value from the SP to echo back and no URL to try to redirect a user-agent to. Assuming validation doesn't expect it to be present always on the `LogoutRequest` (which it should not), this can and should be removed.